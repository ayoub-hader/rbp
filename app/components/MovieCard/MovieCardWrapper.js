import styled from 'styled-components';

const MovieCardWrapper = styled.div`
  border-radius: 6px;
  box-shadow: 1px 1px 5px 1px #ddd;
  display: flex;
  flex-direction: row;
  margin: 10px 0;
  padding: 10px;

  .poster {
    display: inline-flex;
    margin-right: 0.5em;

    img {
      width: 170px;
    }
  }

  .details {
    display: flex;
    flex-direction: column;

    .title {
      font-size: 1em;
      font-weight: bold;
    }

    .year {
      font-size: 0.8em;
      text-style: italic;
    }
  }
`;

export default MovieCardWrapper;
