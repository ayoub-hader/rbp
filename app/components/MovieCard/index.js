/**
 *
 * MovieCard
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
// import styled from 'styled-components';
import NoPoster from 'images/NoPoster.png';
import MovieCardWrapper from './MovieCardWrapper';

const MovieCard = ({ poster, title, year }) => {
  const posterImage = poster === 'N/A' ? NoPoster : poster;
  return (
    <MovieCardWrapper>
      <div className="poster">
        <img src={posterImage} alt="no poster available" />
      </div>
      <div className="details">
        <span className="title">{title}</span>
        <span className="year">{year}</span>
      </div>
    </MovieCardWrapper>
  );
};

MovieCard.propTypes = {
  poster: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  year: PropTypes.string.isRequired,
};

export default MovieCard;
