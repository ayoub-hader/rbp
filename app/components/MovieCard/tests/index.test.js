import React from 'react';
import { render } from '@testing-library/react';
import MovieCard from '../index';

describe('<MovieCard />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <MovieCard poster="img/url.jpg" title="Mind Hunters" year="201x" />,
    );

    expect(firstChild).toMatchSnapshot();
    expect(firstChild.querySelector('img').getAttribute('src')).toEqual(
      'img/url.jpg',
    );
  });

  it('should render with no-poster image mock', () => {
    const {
      container: { firstChild },
    } = render(<MovieCard poster="N/A" title="Mind Hunters" year="201x" />);

    expect(firstChild.querySelector('img').getAttribute('src')).toEqual(
      'IMAGE_MOCK',
    );
  });
});
