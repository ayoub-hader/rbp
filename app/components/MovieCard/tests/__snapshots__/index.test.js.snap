// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCard /> should render and match the snapshot 1`] = `
<div
  class="MovieCardWrapper-nltufh-0 fbJMhn"
>
  <div
    class="poster"
  >
    <img
      alt="no poster available"
      src="img/url.jpg"
    />
  </div>
  <div
    class="details"
  >
    <span
      class="title"
    >
      Mind Hunters
    </span>
    <span
      class="year"
    >
      201x
    </span>
  </div>
</div>
`;
